api/rosetta/data_network_integration_test.go

```diff
@@ -0,0 +1,522 @@
+		},
+		{
+			name: "missing blockchain",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: "",
+					Network:    dps.FlowTestnet.String(),
+				},
+			},
+
+			wantError:                   true,
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "blockchain identifier: blockchain field is missing",
+		},
+		{
+			name: "invalid blockchain",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: invalidBlockchain,
+					Network:    dps.FlowTestnet.String(),
+				},
+			},
+
+			wantError:                   true,
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidNetwork,
+			wantRosettaErrorDescription: fmt.Sprintf("invalid network identifier blockchain (have: %s, want: %s)", invalidBlockchain, dps.FlowBlockchain),
+			wantRosettaErrorDetails:     map[string]interface{}{"blockchain": invalidBlockchain, "network": dps.FlowTestnet.String()},
+		},
+		{
+			name: "missing network",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    "",
+				},
+			},
+
+			wantError:                   true,
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "blockchain identifier: network field is missing",
+		},
+		{
+			name: "invalid network",
+			request: rosetta.StatusRequest{
+				NetworkID: identifier.Network{
+					Blockchain: dps.FlowBlockchain,
+					Network:    invalidNetwork,
+				},
+			},
+
+			wantError:                   true,
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidNetwork,
+			wantRosettaErrorDescription: fmt.Sprintf("invalid network identifier network (have: %s, want: %s)", invalidNetwork, dps.FlowTestnet.String()),
+			wantRosettaErrorDetails:     map[string]interface{}{"blockchain": dps.FlowBlockchain, "network": invalidNetwork},
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			rec, ctx, err := setupRecorder(statusEndpoint, test.request)
+			require.NoError(t, err)
+
+			err = api.Status(ctx)
+			// if it's an error test case, verify we failed correctly
+			if test.wantError {
+				assert.Error(t, err)
+
+				echoErr, ok := err.(*echo.HTTPError)
+				require.True(t, ok)
+
+				assert.Equal(t, test.wantStatusCode, echoErr.Code)
+				gotErr, ok := echoErr.Message.(rosetta.Error)
+				require.True(t, ok)
+
+				assert.Equal(t, test.wantRosettaError, gotErr.ErrorDefinition)
+				assert.Equal(t, test.wantRosettaErrorDescription, gotErr.Description)
+				assert.Equal(t, test.wantRosettaErrorDetails, gotErr.Details)
+
+				return
+			}
```

Comment:
This makes me think that we could actually make our own `assert.ErrorAssertionFunc` for checking rosetta errors, and use that directly. WDYT?

This way, `wantError` can be a `assert.ErrorAssertionFunc` instead of a bool and the test would be much simpler:

```go
			if !test.wantError(t, err) {
				return
			}

			assert.Equal(t, test.wantStatusCode, rec.Result().StatusCode)

			// unpack response
			var status rosetta.StatusResponse
			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &status))

			// verify current block
			assert.Equal(t, status.CurrentBlockID.Index, lastBlock.Height)
			assert.Equal(t, status.CurrentBlockID.Hash, lastBlock.ID().String())
			assert.Equal(t, status.CurrentBlockTimestamp, rosettaTime(lastBlock.Timestamp))

			// verify oldest block
			assert.Equal(t, status.OldestBlockID.Hash, oldestBlockID)
			// this is actually omitted from JSON, due to height being zero, and JSON having the 'omitempty' tag
			assert.Equal(t, status.OldestBlockID.Index, uint64(0))
```

We could make a func that returns a func to inject the error details/description/definition for cases where we want an error, and simply `assert.NoError` otherwise:

```go
func checkRosettaError(statusCode int, def meta.ErrorDefinition, desc string, details map[string]interface{}) func(t *testing.T, err error, v ...interface{}) bool {
	assert.Error(t, err)

	echoErr, ok := err.(*echo.HTTPError)
	require.True(t, ok)

	assert.Equal(t, statusCode, echoErr.Code)
	gotErr, ok := echoErr.Message.(rosetta.Error)
	require.True(t, ok)

	assert.Equal(t, def, gotErr.ErrorDefinition)
	assert.Equal(t, desc, gotErr.Description)
	assert.Equal(t, details, gotErr.Details)
}
```

This way both the test cases and the body of the test itself would be significantly smaller and more readable IMO. WDYT @Maelkum ?